tests/test_pytorch_sparse_dataloader_api.py

```diff
@@ -233,26 +263,41 @@ def test_tiledb_pytorch_sparse_data_api_with_diff_number_of_batch_x_y_rows_empty
                 tiledb_dataset = PyTorchTileDBSparseDataset(
                     x_array=x,
                     y_array=y,
+                    batch_size=BATCH_SIZE,
+                    batch_shuffle=batch_shuffle,
                     x_attribute_names=[
                         "features_" + str(attr) for attr in range(num_of_attributes)
                     ],
                     y_attribute_names=[
                         "features_" + str(attr) for attr in range(num_of_attributes)
                     ],
-                    batch_size=BATCH_SIZE,
                 )
-                next(tiledb_dataset.__iter__())
+                # Exhaust iterator
+                while True:
+                    try:
+                        next(tiledb_dataset.__iter__())
+                    except StopIteration:
+                        # if StopIteration is raised, break from loop
+                        break
 
         # Same test without attribute names explicitly provided by the user
         with tiledb.open(tiledb_uri_x) as x, tiledb.open(tiledb_uri_y) as y:
             with pytest.raises(Exception):
                 tiledb_dataset = PyTorchTileDBSparseDataset(
-                    x_array=x, y_array=y, batch_size=BATCH_SIZE
+                    x_array=x,
+                    y_array=y,
+                    batch_size=BATCH_SIZE,
+                    batch_shuffle=batch_shuffle,
                 )
-                next(tiledb_dataset.__iter__())
+                # Exhaust iterator
+                while True:
+                    try:
+                        next(tiledb_dataset.__iter__())
+                    except StopIteration:
+                        break
```

Comment:
Shouldn't this loop be just `for _ in tiledb_dataset: pass`? As it is, it looks like it will loop forever yielding the first item (at least it should loop forever if `PyTorchTileDBSparseDataset` is a well-behaved Sequence).

Reply:
Could be I think. Will check it!